### PR TITLE
telemetry: use absolute yaw for camera attitude

### PR DESCRIPTION
**Pull request by julianoes**
We now have an interface for the absolute yaw angle of a camera
(respectively a gimbal).

FYI @sanderux.

### CODE REVIEW

File: plugins/telemetry/telemetry.h [hunk 1]
```diff
@@ -364,13 +364,19 @@ class Telemetry : public PluginBase
     /**
      * @brief Get the camera's attitude in quaternions (synchronous).
      *
+     * Note this attitude is relative to North, so in the absolute frame
```
Comment (hamishwillee) [inline plugins/telemetry/telemetry.h]: This is OK, but possibly more than needs to be said. I'd suggest for both changes:

> Note that the yaw component of attitude is relative to North (absolute frame).

**Comment by hamishwillee** [comment]
You also need to update the yaw value here: https://docs.dronecore.io/v/develop/en/api_reference/structdronecore_1_1_telemetry_1_1_euler_angle.html

I'd suggest updating the https://docs.dronecore.io/v/develop/en/api_reference/structdronecore_1_1_telemetry_1_1_quaternion.html but I don't really understand it :-0

Would it be useful for You also need to update the yaw value here: https://docs.dronecore.io/v/develop/en/api_reference/structdronecore_1_1_telemetry_1_1_euler_angle.html to include a vehicle-relative value too, that we calculate? I kind of think it would.

**Comment by julianoes** [comment]
@hamishwillee the fields you're pointing to are just the data types and should not specify the coordinate frame as such. So, these structs can be used for relative or absolute angles.

File: plugins/telemetry/telemetry_impl.cpp [hunk 2]
```diff
@@ -391,7 +391,7 @@ void TelemetryImpl::process_mount_orientation(const mavlink_message_t &message)
     Telemetry::EulerAngle euler_angle {
         mount_orientation.roll,
         mount_orientation.pitch,
-        mount_orientation.yaw
+        mount_orientation.yaw_absolute
```
Comment (shakthi-prashanth-m) [inline plugins/telemetry/telemetry_impl.cpp]: This (http://mavlink.org/messages/common#MOUNT_ORIENTATION) has both `yaw` & `yaw_absolute`. Does the link hold message definitions of MAVLink2 ?
Also, if you change member here, we need to change its usage; isn't it ? :)

Comment (julianoes) [reply plugins/telemetry/telemetry_impl.cpp]: Where do we need to change its usage? I'm confused.

**Comment by hamishwillee** [comment]
> @hamishwillee the fields you're pointing to are just the data types and should not specify the coordinate frame as such. So, these structs can be used for relative or absolute angles.

I don't really understand. If you name something with "absolute" in it's title then you are forcing it to use an absolute frame - `mount_orientation.yaw_absolute`. I guess we aren't specifying which frame of course, but it isn't going to be relative to the vehicle.

File: plugins/telemetry/telemetry_impl.cpp (hunk 2 above)
Comment (shakthi-prashanth-m) [reply plugins/telemetry/telemetry_impl.cpp]: Ah OK! So, its just that [yaw_deg](https://github.com/dronecore/DroneCore/blob/742537418b5892bd30df3d58708826760c3dbef0/plugins/telemetry/telemetry.h#L74) of `EulerAngle` will be absolute now ?

**Comment by julianoes** [comment]
@hamishwillee 
The data type used for an Euler angle or a quaternion in DroneCore just represent an attitude without a frame connected to it. They can be used for absolute or relative, depending on what telemetry information is sent/used.

However, the fields `yaw` and `yaw_absolute` sent in mavlink specify the frame.

**Comment by JonasVautherin** [comment]
So from now on, the yaw of the camera is absolute, and we don't provide an interface to the camera yaw relative to the vehicle. Is that right? :-)

**Comment by julianoes** [comment]
@JonasVautherin that's right. What's your opinion, would you also want the relative yaw?

**Comment by JonasVautherin** [comment]
Fine for me for now. We can always add the relative yaw later if needed, right?

**Comment by julianoes** [comment]
Yes we can.

**Pull request merged**